torch/testing/_internal/distributed/rpc/dist_autograd_test.py

```diff
@@ -1425,22 +1417,28 @@ def test_backward_node_failure_python_udf(self):
 
             if self.rank == 0:
                 # Wait for rank 2 to die.
-                wait_until_node_failure(2)
-
+                shutdown_error_regex = get_shutdown_error_regex(dist_utils.TEST_CONFIG.rpc_backend_name)
+                wait_until_node_failure(2, shutdown_error_regex)
                 # Shutdown sequence is not very well defined and as a result
                 # we might see any error given by get_shutdown_error_regex().
-                with self.assertRaisesRegex(RuntimeError, get_shutdown_error_regex()):
+                with self.assertRaisesRegex(RuntimeError, shutdown_error_regex):
                     # Run backwards, and validate we receive an error since rank 2 is dead.
                     dist_autograd.backward(context_id, [res.sum()])
 
                 # Tell other nodes RPC is done.
                 for i in range(self.world_size):
                     if i != self.rank and i != 2:
-                        rpc.rpc_sync(
-                            worker_name(i),
-                            DistAutogradTest._set_backward_done,
-                            args=(),
-                        )
+                        # Due to non-graceful shutdown of workers, this RPC may not return successfully.
+                        # For example, the destination worker could process the RPC, exit and begin shutdown, and
+                        # shutdown RPC before responding and satisfying this RPC. Therefore, we swallow possible errors here.
+                        try:
+                            rpc.rpc_sync(
+                                "worker{}".format(i),
+                                DistAutogradTest._set_backward_done,
+                                args=(),
+                            )
+                        except Exception as e:
+                            pass
```

Comment:
do we have a specific error message that we can test?

Reply:
Yes, this should be a RPC timed out error message since we never get a response back from the remote end, since it has shutdown.

Reply:
Will update the diff to add this.